### ISSUE TITLE
New 8K/4320p videos will break NewPipe's video renderer.

### ISSUE THREAD

**Issue by SC1040-TS2** [TeamNewPipe/NewPipe#8344]
### Checklist

- [X] I am able to reproduce the bug with the [latest version](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.23.0.

### Steps to reproduce the bug

1. Have Best Resolution selected for a default player resolution.
2. Attempt to watch the following Chubbyemu video: https://www.youtube.com/watch?v=YDaTuW6DYeQ
3. Make note of the error notification that comes up, and attempt to watch it again.
4. Notice that both the 4320p and 2160p WebM presets do not work, and that only 1440p and lower will play the video correctly.

### Expected behavior

The example video plays its 2160p preset correctly.

### Actual behavior

It does not manage to do so.

## Exception
* __User Action:__ play stream
* __Request:__ Player error[type=ERROR_CODE_DECODER_INIT_FAILED] occurred while playing https://www.youtube.com/watch?v=YDaTuW6DYeQ
* __Content Country:__ US
* __Content Language:__ en
* __App Language:__ en
* __Service:__ YouTube
* __Version:__ 0.23.0
* __OS:__ Linux samsung/starqlteue/starqlteue:10/QP1A.190711.020/G960U1UEU9FVA1:user/release-keys 10 - 29
<details><summary><b>Crash log </b></summary><p>

```
com.google.android.exoplayer2.ExoPlaybackException: MediaCodecVideoRenderer error, index=0, format=Format(1, null, null, video/x-vnd.on2.vp9, null, -1, en, [7680, 4320, -1.0], [-1, -1]), format_supported=NO_EXCEEDS_CAPABILITIES
	at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:575)
	at android.os.Handler.dispatchMessage(Handler.java:103)
	at android.os.Looper.loop(Looper.java:237)
	at android.os.HandlerThread.run(HandlerThread.java:67)
Caused by: com.google.android.exoplayer2.mediacodec.MediaCodecRenderer$DecoderInitializationException: Decoder init failed: OMX.qcom.video.decoder.vp9, Format(1, null, null, video/x-vnd.on2.vp9, null, -1, en, [7680, 4320, -1.0], [-1, -1])
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.maybeInitCodecWithFallback(MediaCodecRenderer.java:1017)
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.maybeInitCodecOrBypass(MediaCodecRenderer.java:546)
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.onInputFormatChanged(MediaCodecRenderer.java:1437)
	at com.google.android.exoplayer2.video.MediaCodecVideoRenderer.onInputFormatChanged(MediaCodecVideoRenderer.java:799)
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.readSourceOmittingSampleData(MediaCodecRenderer.java:950)
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.render(MediaCodecRenderer.java:779)
	at com.google.android.exoplayer2.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:998)
	at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:499)
	... 3 more
Caused by: android.media.MediaCodec$CodecException: Error 0xffffec77
	at android.media.MediaCodec.native_configure(Native Method)
	at android.media.MediaCodec.configure(MediaCodec.java:2023)
	at android.media.MediaCodec.configure(MediaCodec.java:1951)
	at com.google.android.exoplayer2.mediacodec.SynchronousMediaCodecAdapter$Factory.createAdapter(SynchronousMediaCodecAdapter.java:51)
	at com.google.android.exoplayer2.mediacodec.DefaultMediaCodecAdapterFactory.createAdapter(DefaultMediaCodecAdapterFactory.java:130)
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.initCodec(MediaCodecRenderer.java:1095)
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.maybeInitCodecWithFallback(MediaCodecRenderer.java:1006)
	... 10 more

```
</details>
<hr>

### Affected Android/Custom ROM version

Android 10.

### Affected device model

Samsung Galaxy S9.

### Additional information

This may be a duplicate of an instance where a 5K/2880p video could not be handled by the player, but potentially unlike that Issue, not even the 4K/2160p preset could be played, and it might have had an attempted fix by now.

Also, further testing revealed that the Delivery-methods-v2 .APK with the MPEG-DASH codec implementation is also affected, meaning further player and extractor refactoring is necessary.

**Comment by Redirion** [comment]
the error basically says that your device capabilities to handle this codec are exceeded

**Comment by SC1040-TS2** [comment]
> the error basically says that your device capabilities to handle this codec are exceeded
It was brought up anyway due to NewPipe likely also not supporting 4320p video resolutions in terms of extractor and player awareness, which could factor into this. Other videos that have a 2160p resolution for native footage can be selected and played with ease, whereas any higher and the player begins to fail at times.

**Comment by opusforlife2** [comment]
Then please test this with a device that is able to handle such a codec-resolution combo. Otherwise this isn't actionable.

**Comment by litetex** [comment]
> 4. Notice that both the 4320p and 2160p WebM presets do not work, and that only 1440p and lower will play the video correctly.

Device is a Samsung Galaxy S9 which has a maximum resolution of 2960x1440px (1440p).

This is also a duplicate of #8016 → From NewPipe side we can't do anything.

**Comment by SC1040-TS2** [comment]
> > 4. Notice that both the 4320p and 2160p WebM presets do not work, and that only 1440p and lower will play the video correctly.
> 
> Device is a Samsung Galaxy S9 which has a maximum resolution of 2960x1440px (1440p).
> 
> This is also a duplicate of #8016 → From NewPipe side we can't do anything.

I understand that. However, any video with a native resolution of 2160p or 2160p60, such as this one here: https://www.youtube.com/watch?v=EqHpnoV00j4 can still be played at that nagive resolution on this device.

The issue is that for videos with any higher native resolution, such as 2880p or 4320p, codec information for scaling to 2160p cannot be extracted. Native 2160p videos can be extracted on this device, but any higher, no.

Also, to answer Opusforlife's request, I am not likely to be able to acquire such a device any time soon. In fact, newer devices, at least where Samsung Galaxy S series ones are concerned, are not even bothering with 1440p vertical resolutions anymore, instead opting for 1080p with high refresh rate combinations.

**Comment by opusforlife2** [comment]
> [github-actions](https://github.com/apps/github-actions) bot reopened this [yesterday](https://github.com/TeamNewPipe/NewPipe/issues/8344#event-6551758639)

@mhmdanas Wot.

**Comment by opusforlife2** [comment]
I can't find the bot repo so I'll reply here: why is reopening the issue expected? I would only expect the label to be removed.

**Comment by Redirion** [comment]
> 

the physical screen size does not affect the capabilities of your video decoder. It may be able to decode 4k, but not anything higher. Your device reports it is unable to decode 8k. That's it. It is a limitation of the decoder, not the screen size.

**Comment by SC1040-TS2** [comment]
> Then please test this with a device that is able to handle such a codec-resolution combo. Otherwise this isn't actionable.

> > 4. Notice that both the 4320p and 2160p WebM presets do not work, and that only 1440p and lower will play the video correctly.
> 
> Device is a Samsung Galaxy S9 which has a maximum resolution of 2960x1440px (1440p).
> 
> This is also a duplicate of #8016 → From NewPipe side we can't do anything.

> @SC1040-TS2 the physical screen size does not affect the capabilities of your video decoder. It may be able to decode 4k, but not anything higher. Your device reports it is unable to decode 8k. That's it. It is a limitation of the decoder, not the screen size.

Greetings, I have some new information on the subject. Apologies for the quoted text.

After a while, this issue appeared to have resolved itself, and video extraction and playback of the originally linked video at its maximum supported resolution on this device, 2160p preset included, functioned as expected.

I should mention that the F-Droid repository version of NewPipe 0.23.0 successfully extracted this video at its available presets, with F-Droid 1.15 being where it was installed from. The version that did not extract the video and got hung up on the unsupported maximum resolution of 4320p and failed to extract the selectable 2160p preset was the NewPipe Upstream Repository version of 0.23.0, installed from a prior Alpha build of F-Droid.

Either user error is behind this, the video was too new at the time and YouTube re-encoding did not handle it well, or something was wrong with either the Alpha build of F-Droid or the NewPipe Upstream repo build.

**Comment by opusforlife2** [comment]
I would bet on the video being too new.
